### PR TITLE
Add blob function test; add type charset

### PR DESCRIPTION
**Pull request by yjw1268**
## What type of PR is this?

- [x] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [x] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3418

## What this PR does / why we need it:
Add blob ut & bvt test in function length, substring.
Add blob bvt test.
Add charset in type.go.

### CODE REVIEW

**Comment by yjw1268** [comment]
@matrix-meow rerun failed

File: pkg/container/types/types.go [hunk 1]
```diff
@@ -203,6 +203,16 @@ func New(oid T, width, scale, precision int32) Type {
 		Scale:     scale,
 		Precision: precision,
 		Size:      int32(TypeSize(oid)),
+		Charset:   CharsetType(oid),
+	}
+}
+
+func CharsetType(oid T) uint8 {
```
Comment (fengttt) [inline pkg/container/types/types.go:210]: What is the?   ...   You are saying oid == T_blob?

**Review by fengttt — COMMENTED** [review]
What is Charset?

**Comment by yjw1268** [comment]
> What is Charset?

MySQL text and blob are of the same type, but the character set is different, so we need to add the charset param.

File: pkg/container/types/types.go (hunk 1 above)
Comment (fengttt) [inline pkg/container/types/types.go:210]: What does returned value mean?   Please add a comment to this function.

File: pkg/sql/plan/function/builtin/multi/substr_test.go [hunk 2]
```diff
@@ -434,6 +434,211 @@ func TestSubStrUTF(t *testing.T) {
...
+		},
+		{
+			name:       "Test24",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -6, -10, true, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test25",
```
Comment (fengttt) [inline pkg/sql/plan/function/builtin/multi/substr_test.go:622]: It is always better to create a longer string for testing.   Something like 50 chars long.

**Comment by fengttt** [comment]
> 

I still don't understand.   Blob is binary, why does that need a charset?   string/text, sure ...

**Comment by yjw1268** [comment]
> > 
> 
> I still don't understand. Blob is binary, why does that need a charset? string/text, sure ...

This is the MySQL protocol which we have to accept...

**Review by fengttt — APPROVED** [review]

**Review by ouyuanning — APPROVED** [review]

**Review by nnsgmsone — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**